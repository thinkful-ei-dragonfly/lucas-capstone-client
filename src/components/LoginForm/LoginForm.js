import React, { useState, useContext } from 'react'
import Context from '../../contexts/Context'
import AuthApiService from '../../services/auth-api-service'


const LoginForm = props => {
  const [ error, setError ] = useState(null)
  // const [ authorized, setAuthorized ] = useState(false)
  const { setLoggedIn } = useContext(Context)
  

  const handleSubmitJwtAuth = e => {
    e.preventDefault()
    setError(null)

    const { user_name, password } = e.target
    const { location, history } = props

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value
    })
      .then(res => {
        user_name.value = ''
        password.value = ''
        setLoggedIn(true)
        // setAuthorized(true)
        history.push('/boards')
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }
  return (
    <form
      className='LoginForm'
      onSubmit={handleSubmitJwtAuth}
      aria-label="Login Form"
    >
      <div role='alert' >
        {error && <p className='red' role='alertdialog' aria-label="Error Message">{error}</p>}
      </div>
      <div className='user_name'>
        <label htmlFor='LoginForm_user_name'>
          User name
          </label>
        <input
          required
          name='user_name'
          aria-label="user_name field"
          id='LoginForm_user_name'>
        </input>
      </div>
      <div className='password'>
        <label htmlFor='LoginForm_password'>
          Password
          </label>
        <input
          required
          name='password'
          aria-label="password field"
          type='password'
          id='LoginForm_password'>
        </input>
      </div>
      <button
        type='submit'
      >
        Login
        </button>
    </form>
  )
}

export default LoginForm