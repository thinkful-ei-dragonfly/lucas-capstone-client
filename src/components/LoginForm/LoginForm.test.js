import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

import LoginForm from './LoginForm'
configure({ adapter: new Adapter() });
describe('LoginForm component', () => {
  const props = {
    something: 'something'
  }

  it('renders a LoginForm by default', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders a form', () => {
    const form = shallow(<LoginForm/>)
    expect(toJson(form)).toMatchSnapshot()
  })
  it('renders a user_name label in the form', () => {
    const user_name = shallow(<LoginForm />)
      .find('.user_name')
    expect(toJson(user_name)).toMatchSnapshot()
  })
  it('renders a password label in the form', () => {
    const password = shallow(<LoginForm />)
      .find('.password')
    expect(toJson(password)).toMatchSnapshot()
  })
})
