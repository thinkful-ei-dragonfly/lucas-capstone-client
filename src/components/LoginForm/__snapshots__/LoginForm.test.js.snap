// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders a LoginForm by default 1`] = `
<form
  aria-label="Login Form"
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm_user_name"
    >
      User name
    </label>
    <input
      aria-label="user_name field"
      id="LoginForm_user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm_password"
    >
      Password
    </label>
    <input
      aria-label="password field"
      id="LoginForm_password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm component renders a form 1`] = `
<form
  aria-label="Login Form"
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm_user_name"
    >
      User name
    </label>
    <input
      aria-label="user_name field"
      id="LoginForm_user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm_password"
    >
      Password
    </label>
    <input
      aria-label="password field"
      id="LoginForm_password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm component renders a password label in the form 1`] = `
<div
  className="password"
>
  <label
    htmlFor="LoginForm_password"
  >
    Password
  </label>
  <input
    aria-label="password field"
    id="LoginForm_password"
    name="password"
    required={true}
    type="password"
  />
</div>
`;

exports[`LoginForm component renders a user_name label in the form 1`] = `
<div
  className="user_name"
>
  <label
    htmlFor="LoginForm_user_name"
  >
    User name
  </label>
  <input
    aria-label="user_name field"
    id="LoginForm_user_name"
    name="user_name"
    required={true}
  />
</div>
`;
